shootout/shootout.py

```diff
@@ -0,0 +1,173 @@
+        default_config = {
+            "cost": 50,
+            "message": "Shoot!",
+            "wait_time": 60,
+            "Session": {"Pot": 0, "Players": [], "Active": False},
+        }
+        self.config.register_guild(**self.default_config)
+
+    async def red_delete_data_for_user(self, **kwargs):
+        """Nothing to delete."""
+        return
+
+    async def game_checks(self, ctx, settings) -> bool:
+        if bool(settings["Session"]["Active"]):
+            await ctx.send("There's already a shootout! Wait for them to finish!")
+            return False
+        if ctx.author.id in settings["Session"]["Players"]:
+            await ctx.send("You're already waiting for the shootout!")
+            return False
+        try:
+            await bank.withdraw_credits(ctx.author, settings["cost"])  # <- Might raise a ValueError!
+            return True
+        except ValueError:
+            currency = await bank.get_currency_name(ctx.guild)
+            await ctx.send("Insufficient funds! This game requires {} {}".format(settings["cost"], currency))
+            return False
+
+    async def add_player(self, ctx, cost):
+        current_pot = await self.config.guild(ctx.guild).Session.Pot()
+        await self.config.guild(ctx.guild).Session.Pot.set(value=(current_pot + cost))
+
+        async with self.config.guild(ctx.guild).Session.Players() as players:
+            players.append(ctx.author.id)
+            num_players = len(players)
+
+        if num_players == 1:
+            wait = await self.config.guild(ctx.guild).wait_time()
+            await ctx.send(
+                "{0.author.mention} is gathering players for a shootout!\n"
+                "Type `{0.prefix}shootout` to join in!\n"
+                "The round will start in {1} seconds.".format(ctx, wait)
+            )
+            await asyncio.sleep(wait)
+            await self.start_game(ctx)
+        else:
+            await ctx.send("{} joined the shootout!".format(ctx.author.mention))
+
+    async def start_game(self, ctx):
+        settings = await self.config.guild(ctx.guild).Session.all()
+        players = [ctx.guild.get_member(player) for player in settings["Players"]]
+        filtered_players = [player for player in players if isinstance(player, discord.Member)]
+        if len(filtered_players) < 2:
+            try:
+                await bank.deposit_credits(ctx.author, settings["Pot"])
+            except BalanceTooHigh as e:
+                await bank.set_balance(ctx.author, e.max_balance)
+            await ctx.send("A shootout with yourself means just shooting at some bottles. For that, no charge.")
+            await self.config.guild(ctx.guild).Session.clear()
+            return
+        listen_message = await self.config.guild(ctx.guild).message()
+        await self.config.guild(ctx.guild).Session.Active.set(True)
+        await ctx.send(
```

Comment:
So, a few things. This should send out this message, but instead of people just typing the message, it should say something like: "When I say draw! type \<msg\> to shoot!" and then randomize a time to wait in some range, then say "Draw!" and thats when the players type the \<msg\>. The issue with the way it is now is that it always waits the same amount of time and becomes predictable.

Reply:
Added delay and fuzzy config values to add a somewhat randomized wait time between when the lobby ends and the game begins.